### PR TITLE
009 Script for requirement [PerformAudioPassThru] SDL must transfer reque…

### PR DESCRIPTION
**Pull request by maria-mih**
…st to HMI in case of valid audioPassThruIcon param

009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua 
All common functions used available at #703
Testing coverage available at #643

@istoimenova Please review

### CODE REVIEW

**Comment by istoimenova** [comment]
@maria-mih, please reference script in the TestSet.

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+-- as well as another related to request valid params
+-- SDL must transfer UI.PerformAudioPassThru (<audioPassThruIcon>, other params)_request + Speak_request (depends on parameters provided by the app) to HMI
+--1. Used preconditions
+--1. All params used in PerformAudioPassThru_request are present and within bounds, audioPassThruIcon is sent with imageType = "DYNAMIC"
+--2. AudioPassThruIcon exists at apps sub-directory of AppStorageFolder (value from ini file)
+--2. Performed steps
+-- Send PerformAudioPassThru (audioPassThruIcon, other params) from mobile to SDL and check:
+--2.1 SDL sends UI.PerformAudioPassThru (audioPassThruIcon (imageType = "DYNAMIC"), other params) to HMI
+--2.2 SDL sends TTS.Speak to HMI
+--2.3 HMI sends UI.PerformAudioPassThru (SUCCESS) to SDL
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @maria-mih, points 2.3-2.4 are redundant in description.

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+--2.4 HMI sends TTS.Speak (SUCCESS) to SDL
+-- Expected result:
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @maria-mih , expected result should be:
SDL sends UI.PerformAudioPassThru (audioPassThruIcon, other params) to HMI
SDL sends TTS.Speak to HMI

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+-- SDL sends PerformAudioPassThru (SUCCESS) to mobile app
+
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @maria-mih, please remove empty row.

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+---------------------------------------------------------------------------------------------
+
+--[[ General configuration parameters ]]
+config.deviceMAC = "12ca17b49af2289436f303e0166030a21e525d266e209267433801a8fd4071a0"
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @maria-mih, please remove space before brackets. Apply this for all calls of functions / require.

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+commonSteps:PutFile("Precondition_PutFile_With_Icon", "icon.png")
+
+function Test:Precondition_Check_audioPassThruIcon_Existence()
+  testCasesForPerformAudioPassThru:Check_audioPassThruIcon_Existence(self)
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @maria-mih, please move parameters on previous row.

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:TestStep_PerformAudioPassThru_AllParameters_DYNAMIC_ImageType_SUCCESS()
+  testCasesForPerformAudioPassThru:PerformAudioPassThru_DYNAMIC_image_SUCCESS_all (self)
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @maria-mih, as the idea of atomic script is to show what exactly is happening, will be better to declare the TestStep in script, not to call common function.
Calling of common functions is useful as preconditions/postconditions.

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+end
+
+function Test:Precondition_ActivateApp()
+  testCasesForPerformAudioPassThru:ActivateAppDiffPolicyFlag
+  (self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+function Test:TestStep_PerformAudioPassThru_AllParameters_DYNAMIC_ImageType_SUCCESS()
```
Comment (istoimenova) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @maria-mih , please be sure that RPC PerformAudioPassThru is allowed by policy.

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua (hunk 3 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @istoimenova Updated at commit b65e19ddf7c55b50ef0f0f4a15c24b3ae659a0b2

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua (hunk 4 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @istoimenova Updated at commit b65e19ddf7c55b50ef0f0f4a15c24b3ae659a0b2

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua (hunk 7 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @istoimenova Updated at commit b65e19ddf7c55b50ef0f0f4a15c24b3ae659a0b2

File: test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua (hunk 6 above)
Comment (maria-mih) [inline test_scripts/API/PerformAudioPassThru/009_ATF_P_PerformAudioPassThru_SUCCESS_Dynamic_Image_All_Params.lua]: @istoimenova Updated at commit b65e19ddf7c55b50ef0f0f4a15c24b3ae659a0b2

**Comment by maria-mih** [comment]
@istoimenova Script has been updated after first review iteration. Please check on commit b65e19ddf7c55b50ef0f0f4a15c24b3ae659a0b2

**Pull request merged**